src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -444,6 +490,25 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
     }
 
     /**
+     * Checks if right curly is alone on line in token the lambda.
+     * @param details for validation.
+     * @return true, if right curly is alone on line.
+     */
+    private static boolean isAloneInLambda(Details details) {
+        final DetailAST lcurly = details.lcurly;
+        final DetailAST rcurly = details.rcurly;
+        final int lcurlyNo = lcurly.getLineNo();
+        final int rcurlyNo = rcurly.getLineNo();
+        if (rcurly.getPreviousSibling() == null) {
+            return rcurlyNo != lcurlyNo;
+        }
+        else {
+            return rcurlyNo != lcurlyNo
+                && rcurlyNo != rcurly.getPreviousSibling().getLineNo();
```

Comment:
I am not about cases this method covers, but about two "return" in its design. Should be only one.